### ISSUE TITLE
Prefer cmd line argument parser

### ISSUE THREAD

**Issue by TIPConsulting** [Melodi17/Q-Limits#2]
I see a lot of finicky code for dealing with cmd arguments. There could benefit to using a tool like [commandlineparser](https://github.com/commandlineparser/commandline) or `System.CommandLine` (still in beta)

Using one of those would also simplify the "help menu" todo item

**Comment by Melodi17** [comment]
Is there a way to do it without restructuring everything?
Because I have used command line parsers before and they require alot of work.

**Comment by TIPConsulting** [comment]
It does take some changes, but you shouldn't need too much work for it.  You can build an annotated model and then replace ` Dictionary<string, string> argD` with it.  Its a bit more setup but you get better parsing, an automatic help menu, and you dont need magic strings in your code to look up arguments from a table (though you do lose colored text)

Build a model to store cmd arguments and annotate it.  The annotations map directly to the user-supplied argument list
```
public class CmdOptions
{
    [Option('m', "module", Required = true, HelpText = "Module is used to contact specified destination")]
    public string Module { get; set; }

    [Option('d', "destination", Required = true, HelpText = "Destination is used to set the device that it targets during execution")]
    public string Destination { get; set; }

     //add the rest of the args here
}
```

New main method parses arguments into the model and dispatches to a delegate
```
static void Main(string[] args)
{
    _ = Parser.Default.ParseArguments<CmdOptions>(args)
        .WithParsed(ExecuteEngine)
        .WithNotParsed(errs =>
        {
            //error condition
        });
}

static void ExecuteEngine(CmdOptions options) 
    => your current "main" code goes here now and you dont need "ArgumentParser.ParseString" or the dictionary

```

Now you have a strongly typed config object that does not rely on magic strings for key lookup.  You can easily pass it around instead of ` Dictionary<string, string> argD`

**Comment by Melodi17** [comment]
Thank you so much! I will implement this as soon as I get the time.

**Comment by Melodi17** [comment]
Issue was resolved as of version 1.0.5
